test/apps/iodemo/io_demo.cc

```diff
@@ -918,19 +930,24 @@ class DemoClient : public P2pDemoCommon {
 
     void connect_all(bool force) {
         if (_active_servers.size() == _server_info.size()) {
+            assert(_status == OK);
             // All servers are connected
             return;
         }
 
         if (!force && !_active_servers.empty()) {
             // The active list is not empty, and we don't have to check the
             // connect retry timeout
+            _status = OK;
             return;
         }
 
         double curr_time = get_time();
         if (curr_time < (_prev_connect_time + opts().client_timeout)) {
             // Not enough time elapsed since previous connection attempt
+            if (_status == WAIT_RESPONSES_TIMEOUT) {
```

Comment:
maybe we can remove the status WAIT_RESPONSES_TIMEOUT altogether: if a connection times out, it triggers reconnect. if there are too many reconnects, it will fail with CONN_RETRIES_EXCEEDED